### DISCUSSION TITLE
Huawai R48XXGx support for all flavours of this recitfier?

### DISCUSSION THREAD

**Discussion by mjpalmowski** [mjpalmowski/CAN-BUS-control-R4875G1-with-ESPHome-and-MQTT#5] [General]
This is a space to discuss adapting the code to work with other members of the Huawei R48xxGx family.

**Comment by kobius77** [comment]
just seen that you recently added a R4850G2.YAML - so seems you didn't start this discussion just to ask if there's demand ;)

i'm currently controlling my one R4850G2 with [this](https://github.com/mb-software/esphome-huawei-r4850) but am considering to get a couple more of these - which would make your work on using multiple devices on one bus [as discussed here](https://github.com/mjpalmowski/CAN-BUS-control-R4875G1-with-ESPHome-and-MQTT/discussions/4) quite valuable.. might give it a try soon.

**Reply by mjpalmowski** [reply]
thanks for your comment, If you do give R4850G2.YAML a try (should run fine on your hardware), I would be keen to know if the AMP settings (other parameter settings, hibernate etc) work properly for the R4850G2. You can certainly put several of these on a single CAN-BUS/DC-BUS. I have placed an example YAML file for  a single unit for MQTT control [here: R4850G2-MQTT](https://github.com/mjpalmowski/CAN-BUS-control-R4875G1-with-ESPHome-and-MQTT/blob/main/R4850G2-mqtt-example.yaml)

And for those that just want to select a model and be done:
![Screenshot 2024-12-15 075408](https://github.com/user-attachments/assets/84c5cee9-05b4-43e7-8b38-27009997d732)

[Here is the YAML for the model dropdown version](https://github.com/mjpalmowski/CAN-BUS-control-R4875G1-with-ESPHome-and-MQTT/blob/main/R48xxGx-example.YAML)

[And here is the YAML to control two R48xxGx units, with model-autodetect and auto scaling-factor setting](https://github.com/mjpalmowski/CAN-BUS-control-R4875G1-with-ESPHome-and-MQTT/blob/main/R48xxGx%20autodetect-autoscaling-dual-inverter-example.yaml)
